### ISSUE TITLE
Fix Proxy Support

### ISSUE THREAD

**Issue by ArtemiyFirsov** [restsharp/RestSharp#805]
RestClient.Proxy property does not work at all. Setting a valid WebProxy object and even setting IT IN APP.CONFIG file of the project does not work. Latest version of rest sharp

**Comment by hallem** [comment]
can you post some of your code for repro steps?

**Comment by fknebels** [comment]
I'm running into this issue as well.

            var restRequest = new RestRequest("", Method.POST) { RequestFormat = DataFormat.Json };
            restRequest.AddHeader("X-API-TOKEN", _apiToken);
            restRequest.AddParameter("surveyId", surveyId); 
            restRequest.AddFile("file", Encoding.UTF8.GetBytes(surveyReponseFile.AsEnumerable().ToCsv()), "feedback.csv",
                "text/csv");

            var restClient = new RestClient(_importUrl) { Proxy = _webProxy };
            return restClient.Execute(restRequest).Content;

But the important piece is only the last 2 lines.  We've turned on logging on our proxy and it is never getting called.  this exact proxy is being passed into a System.Net.Http.HttpClient instance and is being honored.

**Comment by bashiransari** [comment]
Same problem here !! Any updates ??

**Comment by ade90036** [comment]
Same here.. !! Updates??

**Comment by ade90036** [comment]
I was able to run the RestSharp client with WebProxy by using this code.
The version of RestSharp i'm using is the following:  Version=105.2.3.0,

```CODE

//create a WebProxy
WebProxy myproxy = new WebProxy("[your proxy address]", [your proxy port number]);
myproxy.BypassProxyOnLocal = false;

var client = new RestClient(baseURL);
client.Proxy = myproxy

```

I've done few tests and it is working..

**Comment by Leon99** [comment]
Doesn't work for me (105.2.3.0, Win10):
`                    var restClient = new RestClient(".../api/")
                    {
                        Proxy = new WebProxy("localhost", 5555) {BypassProxyOnLocal = false}
                    };
`
The only way to make it use the proxy is to set it as a system setting.

**Comment by Novakov** [comment]
Just hit the same problem (RestSharp 106.2.2), however I was able to work around it;

```
var client = new RestClient("http://base");
var client.ConfigureWebRequest(wr => wr.Proxy = new WebProxy("127.0.0.1", 8888) {BypassProxyOnLocal = false});
```

**Comment by alexeyzimarev** [comment]
It is fixed in 106.3
